RecoEgamma/ElectronIdentification/plugins/ElectronMVANtuplizer.cc

```diff
@@ -178,7 +168,15 @@ ElectronMVANtuplizer::ElectronMVANtuplizer(const edm::ParameterSet& iConfig)
   nValMaps_              (valMapBranchNames_.size()),
   mvaCatTags_            (iConfig.getUntrackedParameter<std::vector<std::string>>("eleMVACats")),
   mvaCatBranchNames_     (iConfig.getUntrackedParameter<std::vector<std::string>>("eleMVACatLabels")),
-  nCats_                 (mvaCatBranchNames_.size())
+  nCats_                 (mvaCatBranchNames_.size()),
+  src_                   (consumes<edm::View<reco::GsfElectron> >(iConfig.getParameter<edm::InputTag>("src")),
+                          consumes<edm::View<reco::GsfElectron> >(iConfig.getParameter<edm::InputTag>("srcMiniAOD"))),
```

Comment:
shouldn't these be mayConsume?

Reply:
I don't know, I just took it from how it was before. What are the guidelines for using `mayConsume` or `consume` Sorry for not addressing your comments before, I somehow forgot about that!

Reply:
One should use `consumes` if everytime the module is called it will always ask for the data. `mayConsume` are used in the cases where sometimes the data is not going to be used.

If there are configurations of the module where one knows a data product will never be requested, then one should not call `consumes` or `mayConsumes`.